search/tests/test_indexing.py

```diff
@@ -36,3 +50,79 @@ def test_strip_xml(self):
         self.resource.save()
         self.assertTrue(self.count_results("you're it") == 1)
         self.assertTrue(self.count_results(xml) == 0)
+
+    def test_sorting(self):
+        """Test sorting for search"""
+        # remove the default resource to control the environment
+        self.resource.delete()
+        # create some resources
+        res1 = self.create_resource(**dict(
+            resource_type="example",
+            title="silly example 1",
+            content_xml="<blah>blah 1</blah>",
+            mpath="/blah1",
+            xa_nr_views=1001,
+            xa_nr_attempts=99,
+            xa_avg_grade=9.9
+        ))
+        res2 = self.create_resource(**dict(
+            resource_type="example",
+            title="silly example 2",
+            content_xml="<blah>blah 2</blah>",
+            mpath="/blah2",
+            xa_nr_views=1002,
+            xa_nr_attempts=98,
+            xa_avg_grade=6.8
+        ))
+        res3 = self.create_resource(**dict(
+            resource_type="example",
+            title="silly example 3",
+            content_xml="<blah>blah 3</blah>",
+            mpath="/blah3",
+            xa_nr_views=1003,
+            xa_nr_attempts=101,
+            xa_avg_grade=7.3
+        ))
+        self.assertEqual(self.count_results(''), 3)
+        # sorting by number of views
+        results = self.search(
+            '',
+            sorting=LoreSortingFields.SORT_BY_NR_VIEWS[0]
+        )
+        # expected position (res3, res2, res1)
+        self.check_resources_sorting(
+            resources=results,
+            nr_views=(res3.xa_nr_views, res2.xa_nr_views, res1.xa_nr_views),
+            nr_attempts=(
+                res3.xa_nr_attempts, res2.xa_nr_attempts, res1.xa_nr_attempts),
+            avg_grades=(
+                res3.xa_avg_grade, res2.xa_avg_grade, res1.xa_avg_grade)
+        )
+        # sorting by number of attempts
+        results = self.search(
+            '',
+            sorting=LoreSortingFields.SORT_BY_NR_ATTEMPTS[0]
+        )
+        # expected position (res3, res1, res2)
+        self.check_resources_sorting(
```

Comment:
I think that a test of the whole sequence is needed: this is the only place where we will ever do it.